ingester/src/data.rs

```diff
@@ -956,6 +1069,22 @@ impl DataBuffer {
         Ok(())
     }
 
+    /// Return a QueryableBatch of the persisting batch after applying new tombstones
+    pub fn get_persisting_data(&self) -> Result<Option<QueryableBatch>> {
+        let persisting = match &self.persisting {
+            Some(p) => p,
+            None => return Ok(None),
+        };
+
+        // persisting data
+        let mut queryable_batch = (*persisting.data).clone();
+
+        // Add new tombstones if any
+        queryable_batch.add_tombstones(&mut self.deletes_during_persisting.clone());
```

Comment:
To go with the change in ingester/src/query.rs, this can be:

```suggestion
        queryable_batch.add_tombstones(&self.deletes_during_persisting);
```